cmd/tf_operator/app/server.go

```diff
@@ -103,7 +98,7 @@ func Run(opt *options.ServerOption) error {
 
 	rl := &resourcelock.EndpointsLock{
 		EndpointsMeta: metav1.ObjectMeta{
-			Namespace: namespace,
+			Namespace: metav1.NamespaceDefault,
```

Comment:
Is NamespaceDefault the default namespace? Why do you use the default Namespace and not the pod's namespace?

Reply:
Shouldn't we be using the namespace that the controller is deployed in for the resource lock? What if we want to deploy different instances of the controller in different namespaces? Wouldn't the end up trying to share the same resource lock causing problems?

Reply:
Why do we need this resource lock? to make sure that there is only one instance to create the controller in the namespace at any time?

Reply:
@jlewi I'm not sure about this, will take a look at it, and file a [issue](https://github.com/tensorflow/k8s/issues/352) to follow up.

@gaocegege It is used for HA.

Reply:
I'd be ok with getting rid of MY_POD_NAME in this PR but leaving MY_POD_NAMESPACE until we can resolve the issue.